site/docs/serialization/binary_serialization.md

```diff
@@ -1,5 +1,63 @@
 # Binary Serialization
 
+Substrait can be serialized into a [protobuf](https://developers.google.com/protocol-buffers) based binary representation. The proto schema/idl files can be found on [GitHub](https://github.com/substrait-io/substrait/tree/main/binary). Proto files are place in the `io.substrait` namespace for C++/Java and the `Substrait.Protobuf` namespace for C#. 
+
+
+## Plan
+
+The main top-level object used to communicate a Substrait plan using protobuf is a Plan message. The plan message is composed of a set of data structures that minimize repetition in the serialization along with one (or more) Relation trees. 
+
+=== "Plan Message"
+
+    ```proto
+%%% proto.message.Plan %%%
+    ```
+
+## Extensions
+Protobuf supports both [simple](/extensions/#simple-extensions) and [advanced](/extensions/#advanced-extensions) extensions. Simple extensions are declared at the plan level and advanced extensions are declared at multiple levels of messages within the plan.
+
+### Simple Extensions
+
+For simple extensions, a plan references the URIs associated with the simple extensions to provide additional plan capabilities. These URIs will list additional relevant information for the plan. 
+
+Since references to simple extensions are used extensively in most plans, the serialized representation is normalized within the plan message type minimize space and serialization costs. For each simple extension YAML referenced, the plan will defined an ExtensionURI object and in-plan surrogate key that can be used in each simple extension extension definition. 
+
+=== "Advanced Extension Protobuf"
+    ```proto
+%%% proto.message.SimpleExtensionURI %%%
+    ```
+
+Once the YAML file is defined with a surrogate key, the surrogate key will be referenced by zero or more `SimpleExtensionDefinition`s. For each simple extension definition, a surrogate key is defined for that specific extension entity. This surrogate key is then pointed to within lower-level primitives (functions, etc) to reference that specific extension. Message properties are named `*_surrogate_key` where the surrogate key is defined and `*_pointer` when referencing the surrogate key. For example `function_surrogate_key` and `function_pointer`.
+
+=== "Advanced Extension Protobuf"
+    ```proto
+%%% proto.message.SimpleExtensionDefinition %%%
+    ```
+
+!!! note
+  Surrogate keys only have meaning within a single plan for the purposes of plan size reduction. They are not some form of global identifier. Different plans may use different surrogate keys for the same specific functions, types, type variations, etc.
+
+!!! note
+  It is valid for a plan to include `SimpleExtensionURI`s and/or `SimpleExtensionDefinition`s that are not referenced directly. It is an optimization to keep this list to the minimum set required.
+
+
+
+### Advanced Extensions
```

Comment:
Is there more going in this section eventually?

Reply:
Yes, definitely. We should add examples and whatnot. In part, this patch introduces the distinction between simple and advanced extensions which felt like an improvement in how we can explain these to people. I thought it was worthwhile getting it up with this limited advanced section to at least give people a framework for how to think about things.